emcc.py

```diff
@@ -2028,10 +2036,8 @@ def get_final():
 
     with ToolchainProfiler.profile_block('memory initializer'):
       memfile = None
-      # for the wasm backend, use a memfile exactly when using pthreads (until
-      # we can remove this temporary hack)
       if (not shared.Settings.WASM_BACKEND and (shared.Settings.MEM_INIT_METHOD > 0 or embed_memfile(options))) or \
-         (shared.Settings.WASM_BACKEND and shared.Settings.USE_PTHREADS):
+         (shared.Settings.WASM_BACKEND and not shared.Settings.MEM_INIT_IN_WASM):
```

Comment:
Yeah, it gets slightly more complicated here with wasm2js, which is another cause for needing a mem init file alongside pthreads. I agree that's not optimal. But I think when fastcomp goes away this will get a lot simpler.